### PR TITLE
ENG-1785 add cypress commands and user management page tests

### PR DESCRIPTION
**Pull request by alepintus**
Hi @avdev4j  @gidesan ,
could you please have a look at this Draft PR and let me know WDYT before creating the PR?

Thanks

### CODE REVIEW

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **alepintus**

If they are not already assigned, you can assign the PR to them by writing `/assign @alepintus` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fapp-builder).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/entando/app-builder/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: src/ui/common/form/RenderRadioInput.js
Comment (gidesan) [inline src/ui/common/form/RenderRadioInput.js:30]: **Never** pass `data-testid` as a prop. It must be totally separated from Component interface

**Comment by entando-jx** [comment]
@alepintus: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | a0442e93c41088ae0737d7daff614743c590b2f4 | [link]() | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Review by gidesan — CHANGES_REQUESTED** [review]

File: test/cypress/integration/users-management.spec.js
Comment (avdev4j) [inline test/cypress/integration/users-management.spec.js:1]: I think we should also move selectors to the support folder.

File: test/cypress/integration/users-management.spec.js
Comment (avdev4j) [inline test/cypress/integration/users-management.spec.js:70]: We should avoid selecting by HTML element. is there no other option to retrieve this table?

**Review by avdev4j — CHANGES_REQUESTED** [review]

**Comment by avdev4j** [comment]
hi @alepintus 
here is my proposal to organize the tests.

Use one describe per business "action" but use multiple "it()" inside to test multiple scenarios, especially the noncommon ones. 

Of course, this is a proposal and a skeleton. It's not the ideal implementation but something we can discuss.

wdyt @gidesan ?

```
describe('Users management', () => {
  beforeEach(() => {
    cy.appBuilderLogin();
    cy.openPageFromMenu(['Users', 'Management'], true);
  });

  afterEach(() => {
    cy.appBuilderLogout();
  });

  describe('access to the main page', () => {
     it('should display the right components', () => {
     });
   });

  describe('access to the user profile', () => {
     // We can use the admin user unless we want specific tests for non admin users
     it('should display the right components in view mode (merge and go back here?)', () => {
     });
     it('should display the right components in edit mode', () => {
     });
   });

  describe('Adding a new user ', () => {
    it('should fire an error with an empty password', () => {
    });
    it('should fire an error with empty...', () => {
    });
    it('should create a user with a valid form', () => {
      // call a generic method to create a user
      // delete the user
    });
  });

  describe('Update an existing user ', () => {
    it('should not be possible on the admin user...', () => {
    });
    it('should fire an error with empty...', () => {
    });
    it('should update the user and his profile with a valid form', () => {
      // create the user
      // update it
      // update the profile
      // delete the user
    });
  });

  describe('Delete a user ', () => {
    it('should not be possible for the current user', () => {
    });
    it('should not be possible for the admin user', () => {
    });
    it('should delete the user', () => {
      // create the user
      // delete the user
    });
  });

  describe('Search a user ', () => {
    it('should display a message when the user has not be found', () => {
    });
    it('should display the user in the result list with admin', () => {
    });
  });
});
```

**Comment by entando-jx** [comment]
@alepintus: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | af32324a4a87fc0f6b02b4021addc6366a73fdf7 | [link]() | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by alepintus** [comment]
Hi @gidesan @avdev4j ,
thanks for the code review. I removed the data-testid from fields renderers and re-organized tests as suggested.
Could you please have a look?

Thanks

**Comment by entando-jx** [comment]
@alepintus: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | af32324a4a87fc0f6b02b4021addc6366a73fdf7 | [link]() | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by entando-jx** [comment]
@alepintus: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | 92be94f8572279f4ae350d7a04edf5da16fce301 | [link]() | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: src/ui/common/form/RenderRadioInput.js
Comment (gidesan) [inline src/ui/common/form/RenderRadioInput.js:48]: still no better alternative to one more div, correct?

File: src/ui/common/form/SwitchRenderer.js
Comment (gidesan) [inline src/ui/common/form/SwitchRenderer.js:18]: I'd avoid this coupling of dataTestId with component logic. Even a wrapping div with data-testid is better.

File: src/ui/common/modal/GenericModal.js
Comment (gidesan) [inline src/ui/common/modal/GenericModal.js:15]: constant missing

File: src/ui/user-profile/common/UserProfileForm.js
Comment (gidesan) [inline src/ui/user-profile/common/UserProfileForm.js:261]: missing constant

File: src/ui/user-profile/common/UserProfileForm.js
Comment (gidesan) [inline src/ui/user-profile/common/UserProfileForm.js:282]: missing constant

File: src/ui/users/authority/UserAuthorityTable.js
Comment (gidesan) [inline src/ui/users/authority/UserAuthorityTable.js:59]: missing constant

File: src/ui/users/authority/UserAuthorityTable.js
Comment (gidesan) [inline src/ui/users/authority/UserAuthorityTable.js:102]: missing constant

File: src/ui/users/authority/UserAuthorityTable.js
Comment (gidesan) [inline src/ui/users/authority/UserAuthorityTable.js:119]: missing constant

File: src/ui/users/common/AddAuthorityModal.js
Comment (gidesan) [inline src/ui/users/common/AddAuthorityModal.js:13]: missing constant

File: src/ui/users/common/AddAuthorityModal.js
Comment (gidesan) [inline src/ui/users/common/AddAuthorityModal.js:42]: missing constant

File: src/ui/users/common/AddAuthorityModal.js
Comment (gidesan) [inline src/ui/users/common/AddAuthorityModal.js:59]: missing constant

File: src/ui/users/common/UserAuthorityPageForm.js
Comment (gidesan) [inline src/ui/users/common/UserAuthorityPageForm.js:53]: missing constant

File: test/cypress/integration/pages-management.spec.js
Comment (gidesan) [inline test/cypress/integration/pages-management.spec.js:12]: please remove commented code

File: test/cypress/support/command/command.js
Comment (gidesan) [inline test/cypress/support/command/command.js:6]: instead of creating from scratch new commands like `getByTestId` and the `getByText` ones, better relying on `cypress-testing-library` which provides them out of the box and makes devs using the same function signatures we already use in `react-testing-library`.

File: test/cypress/integration/users-management-ui.spec.js
Comment (gidesan) [inline test/cypress/integration/users-management-ui.spec.js:10]: did you check if it's possible to use absolute imports? it's appbuilder's codebase standard, not sure if it's possible to do that between test and src

File: test/cypress/integration/users-management-ui.spec.js
Comment (gidesan) [inline test/cypress/integration/users-management-ui.spec.js:24]: I would move this "validate UI" part into the react-testing-library codebase, since it's not so related with the flow. @avdev4j what do you think?

File: test/cypress/integration/users-management.spec.js
Comment (gidesan) [inline test/cypress/integration/users-management.spec.js:82]: please remove commented code

File: test/cypress/integration/users-management.spec.js
Comment (gidesan) [inline test/cypress/integration/users-management.spec.js:83]: please remove commented code

File: test/cypress/integration/users-management.spec.js
Comment (gidesan) [inline test/cypress/integration/users-management.spec.js:98]: please remove commented code

File: test/cypress/integration/users-management.spec.js
Comment (gidesan) [inline test/cypress/integration/users-management.spec.js:118]: please remove commented code

File: test/cypress/integration/users-roles.spec.js
Comment (gidesan) [inline test/cypress/integration/users-roles.spec.js:13]: please remove commented code (but I'm not sure what users-roles will contain then)

File: test/cypress/support/command/user-command.js
Comment (gidesan) [inline test/cypress/support/command/user-command.js:51]: is that a TODO?

**Review by gidesan — CHANGES_REQUESTED** [review]

File: test/cypress/support/command/user-command.js
Comment (alepintus) [reply test/cypress/support/command/user-command.js:51]: Hi @gidesan ,
Thanks for the comment. About the question, I think yes, but this comment was originally added by @avdev4j so we have to ask to him. 
@avdev4j is this a TODO?

File: src/ui/common/form/SwitchRenderer.js
Comment (alepintus) [reply src/ui/common/form/SwitchRenderer.js]: if removed. Thanks

**Review by gidesan — APPROVED** [review]

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/entando/app-builder/pull/1029#pullrequestreview-619730977" title="Approved">gidesan</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fapp-builder).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/entando/app-builder/blob/master/OWNERS)~~ [gidesan]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: test/cypress/support/command/user-command.js
Comment (avdev4j) [reply test/cypress/support/command/user-command.js:51]: I there, It's not a TODO just information about why we have an empty export below. I think we can remove it if we want

**Pull request merged**
